generate_side_by_side.py

```diff
@@ -659,6 +659,22 @@ def build_side_by_side(base_video, new_video, base_ind, new_ind, output_dir, fil
         + [str(pathlib.Path(output_dir, filename))]
     )
 
+def convert_mp4_to_gif(path_to_mp4, path_to_gif, slow_motion=False):
+    fps = 30
+    if slow_motion:
+        fps = 100
+    subprocess.check_output(
+        [
+            "ffmpeg",
+            "-i",
+            str(path_to_mp4),
+            "-vf",
+            f"fps={fps},scale=1080:-1:flags=lanczos,split[s0][s1];[s0]palettegen[p];[s1][p]paletteuse",
+            "-loop",
+            "-1",
+        ]
+        + [str(pathlib.Path(path_to_gif))]
```

Comment:
Assume you will always get pathlib.Path as `path_to_gif`, then you only need to call str() here.